### PR TITLE
Setup file download

### PR DESCRIPTION
**Pull request by ekaterinamarinova**
@BornInMay - javax.xml.bind dependency was a requirement for some reason, cron job is working properly, file structure has been updated - pulling from inbound directory; only problem remains setting up a BlobID with prefix - does that mean creating an object of type BlobId that contains the path to the specified file but only up to the required prefix, without the extension (.txt for example)?

### CODE REVIEW

File: src/main/java/com/personaltask/wordcounter/property/yml/GoogleCloudProperties.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package com.personaltask.wordcounter.property.yml;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
```
Comment (BornInMay) [inline src/main/java/com/personaltask/wordcounter/property/yml/GoogleCloudProperties.java]: Do you really need @Configuration and @EnableConfigurationProperties annotations here?

File: pom.xml [hunk 2]
```diff
@@ -7,19 +7,78 @@
...
             <version>${camel.version}</version>
         </dependency>
+
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-quartz2</artifactId>
+            <version>${camel.version}</version>
+        </dependency>
+
+        <dependency>
```
Comment (BornInMay) [inline pom.xml]: What is the reason to add this dependency? Currently, you don't have any XML interactions in the code

File: src/main/java/com/personaltask/wordcounter/route/WordCounterRoute.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.personaltask.wordcounter.route;
+
+import org.apache.camel.builder.RouteBuilder;
+import org.springframework.stereotype.Component;
+
+@Component
+public class WordCounterRoute extends RouteBuilder {
+
+    public void configure() throws Exception {
+        from("quartz2://simpleTimer?cron=0+0/1+12-23+?+*+MON-FRI")
```
Comment (BornInMay) [inline src/main/java/com/personaltask/wordcounter/route/WordCounterRoute.java]: Make the cron expression configurable via the app's yml files

File: src/test/resources/word-counter.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+### Main configuration file
```
Comment (BornInMay) [inline src/test/resources/word-counter.yml]: Do you really need this yml file for the tests?

File: src/test/resources/word-counter-dev.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+##############################
```
Comment (BornInMay) [inline src/test/resources/word-counter-dev.yml]: Do you really need this yml file for the tests?

File: src/test/java/com.personaltask.wordcounter/processor/DownloadProcessorTest.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    private Exchange exchange;
+
+    @Mock
+    private StorageService storageService;
+
+    @InjectMocks
+    private DownloadProcessor downloadProcessor;
+
+    @Test
+    public void testProcess() throws Exception {
```
Comment (BornInMay) [inline src/test/java/com.personaltask.wordcounter/processor/DownloadProcessorTest.java]: What are you testing here?

File: src/main/resources/word-counter-dev.yml [hunk 7]
```diff
@@ -6,4 +6,14 @@ spring:
...
 
 server:
-  port: 8080
\ No newline at end of file
+  port: 8080
+
+google-storage:
+  file-name-prefix: Dan Millan
+  bucket: word-counter-pld/inbound
+  file-destination: C:\Users\ekaterina.marinova\Desktop
```
Comment (BornInMay) [inline src/main/resources/word-counter-dev.yml]: Remove the unused configs and never use full (absolute) system paths. What will happen if you run this app on non-Windows OS? Use a relative path instead

File: src/main/java/com/personaltask/wordcounter/storage/StorageService.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StorageService.class);
+
+    private Storage storage;
+
+    public StorageService(Storage storage) {
+        this.storage = storage;
+    }
+
+    public String downloadFile(String bucket, String fileNamePrefix) throws NoSuchFileException {
```
Comment (BornInMay) [inline src/main/java/com/personaltask/wordcounter/storage/StorageService.java]: You should use the prefix to find a specific file (or files). If there are any files(blobs) with this prefix, you should download them all. In other words, use the prefix to list and filter all files and then download the appropriates of them (implement multiple files download)

File: src/main/resources/word-counter-dev.yml [hunk 9]
```diff
@@ -6,4 +6,17 @@ spring:
...
+  port: 8080
+
+application:
+  cron: 0+0/1+12-23+?+*+MON-FRI
+
+google-storage:
+  file-name-prefix: 'inbound/Dan Millan'
+  file-destination:
+  bucket: word-counter-pld
+  inbound: word-counter-pld/inbound
```
Comment (BornInMay) [inline src/main/resources/word-counter-dev.yml]: You don't need the "word-counter-pld" prefix for each location - inbound, done, outbound and error. "word-counter-pld" is the bucket name. You already have it in the properties and you can use string concatenation to build the full location if you need it.

File: pom.xml [hunk 10]
```diff
@@ -7,19 +7,78 @@
     <groupId>com.personaltask</groupId>
```
Comment (BornInMay) [inline pom.xml]: Remove all unnecessary dependencies, configs and etc from here.

File: src/main/resources/word-counter-dev.yml (hunk 7 above)
Comment (ekaterinamarinova) [reply src/main/resources/word-counter-dev.yml]: fixed

File: src/test/resources/word-counter-dev.yml (hunk 5 above)
Comment (ekaterinamarinova) [reply src/test/resources/word-counter-dev.yml]: fixed

File: src/test/resources/word-counter.yml (hunk 4 above)
Comment (ekaterinamarinova) [reply src/test/resources/word-counter.yml]: fixed

File: src/main/java/com/personaltask/wordcounter/route/WordCounterRoute.java (hunk 3 above)
Comment (ekaterinamarinova) [reply src/main/java/com/personaltask/wordcounter/route/WordCounterRoute.java]: fixed

File: pom.xml (hunk 2 above)
Comment (ekaterinamarinova) [reply pom.xml]: fixed

File: src/main/java/com/personaltask/wordcounter/property/yml/GoogleCloudProperties.java (hunk 1 above)
Comment (ekaterinamarinova) [reply src/main/java/com/personaltask/wordcounter/property/yml/GoogleCloudProperties.java]: fixed

File: src/main/java/com/personaltask/wordcounter/storage/StorageService.java (hunk 8 above)
Comment (ekaterinamarinova) [reply src/main/java/com/personaltask/wordcounter/storage/StorageService.java]: fixed

File: src/main/resources/word-counter-dev.yml (hunk 9 above)
Comment (ekaterinamarinova) [reply src/main/resources/word-counter-dev.yml]: fixed

**Pull request merged**